packages/wallet/src/rollup-state-validator.ts

```diff
@@ -0,0 +1,285 @@
+  TokenType,
+  State,
+  StateUpdate,
+  StateSnapshot,
+  InclusionProof,
+  StateMachineCapacityError,
+  SignatureError,
+  AGGREGATOR_ADDRESS,
+  abiEncodeTransaction,
+  abiEncodeState,
+  parseStateFromABI,
+  DefaultRollupStateMachine,
+  InsufficientBalanceError,
+  NegativeAmountError,
+  InvalidTransactionTypeError,
+  InvalidTokenTypeError,
+  isStateTransitionError,
+  ValidationOutOfOrderError,
+  AggregatorUnsupportedError,
+  DefaultRollupBlock,
+} from './index'
+
+import {
+  RollupBlock,
+  RollupStateValidator,
+  RollupTransitionPosition,
+  FraudCheckResult,
+  RollupStateMachine,
+  RollupTransition,
+  SlippageError,
+  LocalMachineError,
+  LocalFraudProof,
+  UniTokenType,
+  PigiTokenType,
+} from './types'
+
+import { Transaction } from 'ethers/utils'
+import {
+  parseTransitionFromABI,
+  parseTransactionFromABI,
+} from './serialization'
+
+const log = getLogger('rollup-guard')
+export class DefaultRollupStateValidator implements RollupStateValidator {
+  public rollupMachine: DefaultRollupStateMachine
+  public currentPosition: RollupTransitionPosition = {
+    blockNumber: 0,
+    transitionIndex: 0,
+  }
+
+  public static async create(
+    genesisState: State[],
+    stateMachineDb: DB
+  ): Promise<DefaultRollupStateValidator> {
+    const theRollupMachine = (await DefaultRollupStateMachine.create(
+      genesisState,
+      stateMachineDb,
+      ChecksumAgnosticIdentityVerifier.instance()
+    )) as DefaultRollupStateMachine
+    return new DefaultRollupStateValidator(theRollupMachine)
+  }
+
+  constructor(theRollupMachine: DefaultRollupStateMachine) {
+    this.rollupMachine = theRollupMachine
+  }
+
+  public async getCurrentVerifiedPosition(): Promise<RollupTransitionPosition> {
+    return this.currentPosition
+  }
+
+  public async getInputStateSnapshots(
+    transition: RollupTransition
+  ): Promise<StateSnapshot[]> {
+    if (isSwapTransition(transition)) {
+      const swapperSnapshot: StateSnapshot = await this.rollupMachine.getSnapshotFromSlot(
+        transition.senderSlotIndex
+      )
+      const uniSnapshot: StateSnapshot = await this.rollupMachine.getState(
+        UNISWAP_ADDRESS
+      )
+      return [swapperSnapshot, uniSnapshot]
+    } else if (isCreateAndTransferTransition(transition)) {
+      const nextAccountKey: number = this.rollupMachine.getNextNewAccountSlot()
+      const senderSnapshot: StateSnapshot = await this.rollupMachine.getSnapshotFromSlot(
+        transition.senderSlotIndex
+      )
+      const recipientSnapshot: StateSnapshot = await this.rollupMachine.getSnapshotFromSlot(
+        transition.recipientSlotIndex
+      )
+      return [senderSnapshot, recipientSnapshot]
+    } else if (isTransferTransition(transition)) {
+      const senderSnapshot: StateSnapshot = await this.rollupMachine.getSnapshotFromSlot(
+        transition.senderSlotIndex
+      )
+      const recipientSnapshot: StateSnapshot = await this.rollupMachine.getSnapshotFromSlot(
```

Comment:
Can just set senderSlot and recipientSlot in the if / else if / else and then do the `this.rollupMachine.getSnapshotFromSlot(...)` calls and the return once at the end.